### ISSUE TITLE
Random error in CommonTools/RecoAlgos unit test

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#28841]
We have a random error in the unit test of CommonTools/RecoAlgos due to https://github.com/cms-sw/cmssw/blob/master/CommonTools/RecoAlgos/test/FKDTree_t.cpp#L50.
Running 100 times FKDTree_t I got
- 99 times:
```
Running ..

OK (2)
```
- 1 time:
```
Running ..F

FKDTree_t.cpp:107:Assertion
Test name: TestFKDTree::test3D
equality assertion failed
- Expected: 1001
- Actual  : 1000

Failures !!!
Run: 2   Failure total: 1   Failures: 1   Errors: 0
```

@smuzaffar suggested the test will fail is "rand()"  return 0 at
https://github.com/cms-sw/cmssw/blob/master/CommonTools/RecoAlgos/test/FKDTree_t.cpp#L50
https://github.com/cms-sw/cmssw/blob/master/CommonTools/RecoAlgos/test/FKDTree_t.cpp#L98 

cc: @felicepantaleo

**Comment by silviodonato** [comment]
assign reco

**Comment by silviodonato** [comment]
assign reconstruction

**Comment by slava77** [comment]
@silviodonato 
do these still show up?
I'm curious if this got fixed "by itself"

**Comment by silviodonato** [comment]
@slava77 Indeed, I don't see the error in the IB test anymore.
I will try again run it locally many times. Thanks for the reminder

**Comment by silviodonato** [comment]
I tried again running 1000 times, and I got two times:
```
FKDTree_t.cpp:107:Assertion
Test name: TestFKDTree::test3D
equality assertion failed
- Expected: 1001
- Actual  : 1000

Failures !!!
Run: 2   Failure total: 1   Failures: 1   Errors: 0

```

**Comment by silviodonato** [comment]
See https://github.com/cms-sw/cmssw/pull/31379

**Comment by silviodonato** [comment]
solved with #31379
